lib/ext_json.js

```diff
@@ -230,8 +230,13 @@ function serializeDocument(doc, options) {
     } else if (val != null && typeof val === 'object') {
       _doc[name] = serializeDocument(val, options);
     }
-
     _doc[name] = serializeValue(val, options);
+    if (val instanceof RegExp) {
+      if (RegExp.prototype.flags === undefined) {
+        val.flags = val.toString().match(/[gimuy]*$/)[0];
```

Comment:
We shouldn't mutate `val.flags`, since that is a user-defined object. We should do something more like:

```js
let flags = val.flags;
if (flags === undefined) {
  flags = val.toString().match(/[gimuy]*$/)[0]
}
...
```